contrib/zenith/libpagestore.c

```diff
@@ -178,15 +181,75 @@ zenith_call(ZenithRequest request)
 	return (ZenithResponse *) resp;
 }
 
+
+static bool
+check_zenith_timeline(char **newval, void **extra, GucSource source)
+{
+	uint8 ztimelineid[16];
+	return **newval == '\0' || HexDecodeString(ztimelineid, *newval, 16);
+}
+
 /*
  * Module initialization function
  */
 void
 _PG_init(void)
 {
+	DefineCustomStringVariable("zenith.page_server_connstring",
+							   "connection string to the page server",
+							   NULL,
+							   &page_server_connstring,
+							   "",
+							   PGC_POSTMASTER,
+							   0,	/* no flags required */
+							   NULL, NULL, NULL);
+
+	DefineCustomStringVariable("zenith.callmemaybe_connstring",
+							   "Connection string that Page Server or WAL safekeeper should use to connect to us",
+							   NULL,
+							   &callmemaybe_connstring,
+							   "",
+							   PGC_POSTMASTER,
+							   0,	/* no flags required */
+							   NULL, NULL, NULL);
+
+	DefineCustomStringVariable("zenith.zenith_timeline",
+							   "Zenith timelineid the server is running on",
+							   NULL,
+							   &zenith_timeline,
+							   "",
+							   PGC_POSTMASTER,
+							   0,	/* no flags required */
+							   check_zenith_timeline, NULL, NULL);
+
+	DefineCustomBoolVariable("zenith.wal_redo",
+							 "start in wal-redo mode",
+							 NULL,
+							 &wal_redo,
+							 false,
+							 PGC_POSTMASTER,
+							 0,
+							 NULL, NULL, NULL);
+
 	if (page_server != NULL)
 		zenith_log(ERROR, "libpqpagestore already loaded");
 
 	zenith_log(PqPageStoreTrace, "libpqpagestore already loaded");
 	page_server = &api;
+
+	//Is there more correct way to pass CustomGUC to postgres code?
+	zenith_timeline_walproposer = zenith_timeline;
```

Comment:
Why do we need zenith_timeline variable at all? Why not to use zenith_timeline_walproposer directly in DefineCustomStringVariable?